tests/e2e/framework/kubernetes.go

```diff
@@ -718,7 +733,13 @@ func (k *KubeInfo) deployIstio() error {
 		}
 
 		testIstioYaml := filepath.Join(k.TmpDir, "yaml", mcRemoteInstallFile)
-		if err := k.generateRemoteIstio(testIstioYaml, *useAutomaticInjection, *proxyHub, *proxyTag); err != nil {
+		var err error
+		if *splitHorizon {
+			err = k.generateRemoteIstioForSplitHorizon(testIstioYaml, remoteNetworkName, *proxyHub, *proxyTag)
```

Comment:
Similar to my comment above about auth enabled.   It looks like you don't have any provisions to handle both states of useAutomaticInjection.   It would be good to throw an error or at least add a comment on what is compatible at initial checkin time.

Reply:
Fixed in https://github.com/istio/istio/pull/12709/commits/e77f11786cb3145f6d3a9ef3dd336d8396b1d259